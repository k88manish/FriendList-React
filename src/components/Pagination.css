:local(.paginationBar) {
  display: flex;
  height: 35px;
  color: #fff;
}

:local(.listStyle) {
  list-style: none;
  margin: auto;
  padding: 0 0 0 0;
}

:local(.listStyle) li {
  display: inline-block;
  padding: 2px 4px;
  margin: 5px;
  border: 1px solid #f0f0f0;
  border-radius: 5px;
  cursor: pointer;
}

:local(.listStyle) li:hover {
  background: #f0f0f0;
  border: 1px solid #5c75b0;
  color: #5c75b0;
}

:local(.listStyle) :local(.active) {
  background: #f0f0f0;
  border: 1px solid #5c75b0;
  color: #5c75b0;
}
