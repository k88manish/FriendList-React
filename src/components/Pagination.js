import React, { Component } from "react";
import PropType from "prop-types";
import * as constant from "../constants/Common";
import style from "./Pagination.css";

class Pagination extends Component {
  getPages(props) {
    const { recordCount, pageSize, activePage } = props;
    const totalPages = Math.ceil(recordCount / pageSize);
    const startPage = activePage - 2 > 1 ? activePage - 2 : 1;
    const endPage = activePage + 2 > totalPages ? totalPages : activePage + 2;

    const pages = [];
    for (let i = startPage; i <= endPage; i++) {
      pages.push(i);
    }
    return { pages, totalPages };
  }

  onPageBtnClick = page => {
    this.props.loadPage(page);
  };

  onNextOrPrevClick = btn => {
    const { activePage } = this.props;
    if (btn === constant.NEXT) {
      this.props.loadPage(activePage + 1);
    } else if (btn === constant.PREVIOUS) {
      this.props.loadPage(activePage - 1);
    }
  };

  render() {
    const { activePage } = this.props;
    const { pages, totalPages } = this.getPages(this.props);
    return (
      <div className={style.paginationBar}>
        <ul className={style.listStyle}>
          {activePage > 1 && (
            <li className="prev" onClick={() => this.onNextOrPrevClick(constant.PREVIOUS)}>
              {constant.PREVIOUS}
            </li>
          )}
          {pages.map(e => (
            <li key={e} className={e === activePage ? style.active : ""} onClick={() => this.onPageBtnClick(e)}>
              {e}
            </li>
          ))}
          {activePage < totalPages && (
            <li className="next" onClick={() => this.onNextOrPrevClick(constant.NEXT)}>
              {constant.NEXT}
            </li>
          )}
        </ul>
      </div>
    );
  }
}

Pagination.propTypes = {
  activePage: PropType.number.isRequired,
  recordCount: PropType.number.isRequired,
  pageSize: PropType.number.isRequired,
  loadPage: PropType.func.isRequired
};

export default Pagination;
