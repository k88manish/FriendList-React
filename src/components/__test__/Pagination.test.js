import React from "react";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";
import Pagination from "../Pagination";

configure({ adapter: new Adapter() });

const mockPageLoad = jest.fn();
const propsData = {
  activePage: 1,
  recordCount: 4,
  pageSize: 2,
  loadPage: mockPageLoad
};

describe("Pagination Test Suits", () => {
  it("it should render correctly", () => {
    const componentWrapper = shallow(<Pagination {...propsData} />);
    expect(componentWrapper).toMatchSnapshot();
  });

  it("Previous button will be not able there for first page", () => {
    const componentWrapper = shallow(<Pagination {...propsData} />);
    expect(componentWrapper.find('li[children="Prev"]').length).toEqual(0);
  });

  it("Next button click should call the load page function", () => {
    const componentWrapper = shallow(<Pagination {...propsData} />);
    componentWrapper.find('li[children="Next"]').simulate("click");
    expect(mockPageLoad).toHaveBeenCalledTimes(1);
  });

  it("Previous button will be not able there for first page", () => {
    const props = Object.assign({}, propsData, { activePage: 2 });
    const componentWrapper = shallow(<Pagination {...props} />);
    expect(componentWrapper.find('li[children="Next"]').length).toEqual(0);
  });

  it("Previous button will be not able there for first page", () => {
    mockPageLoad.mockReset();
    const props = Object.assign({}, propsData, { activePage: 2 });
    const componentWrapper = shallow(<Pagination {...props} />);
    componentWrapper.find('li[children="Prev"]').simulate("click");
    expect(mockPageLoad).toHaveBeenCalledTimes(1);
  });
});
