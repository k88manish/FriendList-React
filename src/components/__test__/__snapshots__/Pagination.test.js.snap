exports[`Pagination Test Suits it should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    activePage={1}
    loadPage={[Function]}
    pageSize={2}
    recordCount={4} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className={undefined}>
        <li
                className={undefined}
                onClick={[Function]}>
                1
        </li>
        <li
                className=""
                onClick={[Function]}>
                2
        </li>
        <li
                className="next"
                onClick={[Function]}>
                Next
        </li>
</ul>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <li
              className={undefined}
              onClick={[Function]}>
              1
</li>,
            <li
              className=""
              onClick={[Function]}>
              2
</li>,
          ],
          <li
            className="next"
            onClick={[Function]}>
            Next
</li>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 1,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 2,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className={undefined}>
          <li
                    className={undefined}
                    onClick={[Function]}>
                    1
          </li>
          <li
                    className=""
                    onClick={[Function]}>
                    2
          </li>
          <li
                    className="next"
                    onClick={[Function]}>
                    Next
          </li>
</ul>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            Array [
              <li
                className={undefined}
                onClick={[Function]}>
                1
</li>,
              <li
                className=""
                onClick={[Function]}>
                2
</li>,
            ],
            <li
              className="next"
              onClick={[Function]}>
              Next
</li>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 1,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 2,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "next",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Next",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
